app/javascript/survey/views/Response.vue

```diff
@@ -33,26 +47,138 @@
 </template>
 
 <script>
-import Branding from 'shared/components/Branding.vue';
-import Rating from 'survey/components/Rating.vue';
-import CustomButton from 'shared/components/Button';
-import TextArea from 'shared/components/TextArea.vue';
+import Branding from 'shared/components/Branding';
+import Spinner from 'shared/components/Spinner';
+import Rating from 'survey/components/Rating';
+import Feedback from 'survey/components/Feedback';
+import Banner from 'survey/components/Banner';
 import configMixin from 'shared/mixins/configMixin';
+import { getSurveyDetails, updateSurvey } from 'survey/api/survey';
+import alertMixin from 'shared/mixins/alertMixin';
+const BRAND_LOGO = '/brand-assets/logo.svg';
 export default {
   name: 'Home',
   components: {
     Branding,
     Rating,
-    CustomButton,
-    TextArea,
+    Spinner,
+    Banner,
+    Feedback,
   },
-  mixins: [configMixin],
+  mixins: [alertMixin, configMixin],
+  props: {
+    showHomePage: {
+      type: Boolean,
+      default: false,
+    },
+  },
+
   data() {
     return {
-      message: '',
-      isSubmitted: false,
+      surveyDetails: null,
+      isLoading: false,
+      errorMessage: null,
+      selectedRating: null,
+      feedbackMessage: '',
+      isUpdating: false,
+      logo: '',
     };
   },
+  computed: {
+    surveyId() {
+      const pageURL = window.location.href;
+      return pageURL.substr(pageURL.lastIndexOf('/') + 1);
+    },
+    isRatingSubmitted() {
+      return this.surveyDetails && this.surveyDetails.rating;
+    },
+    isFeedbackSubmitted() {
+      return this.surveyDetails && this.surveyDetails.feedback_message;
+    },
+    isButtonDisabled() {
+      return !(this.selectedRating && this.feedback);
+    },
+    enableBanner() {
```

Comment:
Would be nice to see `isSomething` pattern being followed like other functions to keep it consisent.